streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredVersionedKeyValueStore.java

```diff
@@ -89,6 +102,18 @@ private class MeteredVersionedKeyValueStoreInternal
         private final Serde<V> plainValueSerde;
         private StateSerdes<K, V> plainValueSerdes;
 
+        private final Map<Class, QueryHandler> queryHandlers =
+            mkMap(
+                mkEntry(
+                    RangeQuery.class,
```

Comment:
Do we need to add this one? `handleRangeQuery` is not implemented anyway?

Reply:
> Do we need to add this one? `handleRangeQuery` is not implemented anyway?

Yes, we do. The current unit test `shouldThrowOnIQv2RangeQuery` must still pass.

Reply:
Ah I see -- but for this case, we also need to add `KeyQuery` and also keep the existing test.

As mentioned elsewhere, we could add support for `KeyQuery` and `RangeQuery` in a follow up KIP, so for now we should keep the guard for both in place.